### ISSUE TITLE
Faster implementation available for `vectorize()`

### ISSUE THREAD

**Issue by LucaMarconato** [scverse/spatialdata#634]
During the Ghent Hackathon ([BioHackrXiv here](https://osf.io/preprints/biohackrxiv/uf7d3)) a faster implementation for `vectorize()` has been developed by @hey2homie: https://github.com/saeyslab/VIB_Hackathon_June_2024/blob/main/polygons/polygons_test.ipynb.

The implementation also fixes https://github.com/scverse/spatialdata/issues/583, but relies on `opencv`, which is a heavy dependency.

Still, we should see if that code could help improving the performance.

**Comment by LucaMarconato** [comment]
@hey2homie I won't have the time to check your approach soon, but if you think that the same approach could be used without using `opencv` and are willing to open a PR, your contribution would be very welcome 😊

**Comment by hey2homie** [comment]
@LucaMarconato, thanks for reaching out! Was a bit busy lately to follow-up on this. There were some minor issues in the code from the hackathon, but I've already fixed them. I will work a bit more on this to polish and submit PR.

By the way, the speed bust most like coming from not using chunking with `dask` arrays but simply using `numpy` as both `opencv` and `skimage` have almost identical performance. So, abandoning `opencv` in favor to `skimage` wouldn't be a problem!

**Comment by LucaMarconato** [comment]
Thank you for the explanation, and thank you in advance for the PR; it is very appreciated!

**Comment by giovp** [comment]
is this related to #560 ?

**Comment by LucaMarconato** [comment]
Yes, #560 was the original implementation but a faster one has been developed from @hey2homie during the Ghent hackathon.
